atlasdb-config/src/main/java/com/palantir/atlasdb/factory/ServiceDiscoveringAtlasSupplier.java

```diff
@@ -57,8 +58,11 @@ public KeyValueService getKeyValueService() {
     }
 
     public TimestampService getTimestampService() {
-        log.trace("Fetching timestamp service. This should only happen once.",
+        log.trace("Fetching timestamp service from thread {}. This should only happen once.",
+                Thread.currentThread().getName(),
                 new RuntimeException("Not necessarily exceptional, but here's a stack trace..."));
```

Comment:
we can probably get rid of the runtime exception above as below will show all threads, including this one